diesel/src/query_builder/ast_pass.rs

```diff
@@ -46,15 +56,41 @@ impl<'a, DB> AstPass<'a, DB> where
         }
     }
 
+    pub fn push_sql(&mut self, sql: &str) {
+        if let AstPassInternals::ToSql(ref mut builder) = self.internals {
+            builder.push_sql(sql);
+        }
+    }
```

Comment:
Yes. I think we will need to stick `#[inline(always)]` on the impls for tuples and `SelectStatement`, as I believe that both of them are currently considered too expensive to inline and are extreme hot paths. I was planning on leaving that for a separate commit though.